lib/RefBackend/RefBackend.cpp

```diff
@@ -137,23 +154,31 @@ class MungeCallingConventions
     OpBuilder b(module.getBodyRegion());
     DenseMap</*returnElementType*/ Type, FuncOp> consumeFuncReturnFuncs;
     DenseSet<FuncOp> consumeFuncReturnFuncsSet;
-    auto createConsumeFuncReturnFunc = [&](Type elemTy, std::string funcName) {
+    auto createConsumeFuncReturnFunc = [&](Type elemTy, std::string funcName,
+                                           bool isMemrefType = true) {
+      auto returnType = isMemrefType
+                            ? UnrankedMemRefType::get(elemTy, /*memorySpace=*/0)
+                            : elemTy;
```

Comment:
I would pass the `returnType` instead of elemTy and remove the ` bool isMemrefType = true` parameter. IMO, readability would be better.

Reply:
Yup, that Makes sense. Thanks.